paimon-core/src/main/java/org/apache/paimon/append/AppendOnlyTableCompactionCoordinator.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.append;
+
+import org.apache.paimon.annotation.VisibleForTesting;
+import org.apache.paimon.data.BinaryRow;
+import org.apache.paimon.io.DataFileMeta;
+import org.apache.paimon.manifest.FileKind;
+import org.apache.paimon.manifest.ManifestEntry;
+import org.apache.paimon.operation.AppendOnlyFileStoreScan;
+import org.apache.paimon.operation.ScanKind;
+import org.apache.paimon.table.AppendOnlyFileStoreTable;
+import org.apache.paimon.utils.SnapshotManager;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/** {@link AppendOnlyFileStoreTable} compact coordinator. */
+public class AppendOnlyTableCompactionCoordinator {
+    private final SnapshotManager snapshotManager;
+    private final AppendOnlyFileStoreScan scan;
+
+    private Long snapshotId;
+    private final long targetFileSize;
+    private final int minFileNum;
+    private final int maxFileNum;
+
+    private final Map<BinaryRow, PartitionCompactCoordinator> partitionCompactCoordinators =
+            new HashMap<>();
+
+    public AppendOnlyTableCompactionCoordinator(
```

Comment:
Can you modify the `CompactAction` for this non-bucket append-only table?
You can modify topology like `CompactorSourceBuilder`.